src/main/java/hu/wurfel/refference/school/diary/DiaryCrudService.java

```diff
@@ -1,29 +1,33 @@
 package hu.wurfel.refference.school.diary;
 
 import org.jetbrains.annotations.NotNull;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
+import java.util.List;
 
 @Service
 public class DiaryCrudService {
-    @Autowired
-    private DiaryRepository diaryRepository;
 
-    public ArrayList<Diary> getAllDiaries() {
-        return new ArrayList<Diary>(diaryRepository.findAll());
+    private final DiaryRepository diaryRepository;
+
+    public DiaryCrudService(DiaryRepository diaryRepository) {
+        this.diaryRepository = diaryRepository;
+    }
+
+    public List<Diary> getAllDiaries() {
+        return new ArrayList<>(diaryRepository.findAll());
```

Comment:
A redundancia itt is megvan, azt nem írom le megint.
Viszont azt igen, hogy ha findAll repo methodot hívsz, akkor célszerű a service method nevében is jelezni, tehát nem get, hanem find.
Ennek van mélyebb jelentősége is, de az még itt irreleváns.
Mi úgy csináljuk pl, hogy a findById entitással tér vissza, a getById pedig valamilyen DTO-val (ez is általánosítva természetesen). Ez saját konvenció, nem kell követni, viszont ha find a repo method, akkor ennek is célszerű findnak lenni.